integration/restore_test.go

```diff
@@ -0,0 +1,137 @@
+// +build integration
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8upv1a1 "github.com/vshn/k8up/api/v1alpha1"
+	"github.com/vshn/k8up/controllers"
+)
+
+type RestoreTestSuite struct {
+	EnvTestSuite
+
+	GivenRestore *k8upv1a1.Restore
+}
+
+const RestoreName = "restore-integration-test"
+
+func Test_Restore(t *testing.T) {
+	suite.Run(t, new(RestoreTestSuite))
+}
+
+func NewRestoreResource() *k8upv1a1.Restore {
+	return &k8upv1a1.Restore{
+		Spec: k8upv1a1.RestoreSpec{
+			RestoreMethod: &k8upv1a1.RestoreMethod{
+				S3: &k8upv1a1.S3Spec{
+					Bucket:   "backups",
+					Endpoint: "https://s3-endpoint.local",
+				},
+			},
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      RestoreName,
+			Namespace: NS,
```

Comment:
I recommend to use different namespace for each test. can even be generated ones, main point is that they don't interfere with eachother with sideeffects or resources we might not know about at programming time.

Reply:
I'll look into that in the afternoon.

But I believe this might be something that we can delay until it becomes a problem. Because currently, each test suite gets it's own envtest. And – as I understand – every individual test suite is executed sequentially. So if every test of each suite properly cleans up after itself, we won't have a problem soon.

Reply:
I was referencing to each test within the same suite so that they're side effects free within the suite. It might be enough to just use the test name as the namespace

Reply:
I've implemented it, but since garbage collection is not running in envtest it actually provides little value. When removing a NS, it does not remove the respective resources. Same goes for owner references, btw.